nixos/modules/config/networking.nix

```diff
@@ -146,6 +168,45 @@ in
         '';
       };
     };
+
+    networking.services = lib.mkOption {
+      description = "Internet network services list overrides";
+      default = { };
+      example = {
+        netstat = {
+          port = 15;
+          protocols = [ "tcp" ];
+        };
+        qotd = {
+          port = 17;
+          protocols = [ "tcp" ];
+          aliases = [ "quote" ];
+        };
+        msp = {
+          port = 18;
+          protocols = [ "tcp" "udp" ];
+        };
+      };
+      type = with lib.types;
+        attrsOf (submodule {
+          options = {
+            port = lib.mkOption {
+              description = "Port number.";
+              type = int;
+            };
+            protocols = lib.mkOption {
+              description = "Type of protocol. Typical values include tcp and udp.";
+              type = listOf str;
```

Comment:
👍🏼 we can't use enum or so here because there are many more protocols than we can collect here